src/serac/physics/materials/parameterized_solid_functional_material.hpp

```diff
@@ -0,0 +1,168 @@
+// Copyright (c) 2019-2022, Lawrence Livermore National Security, LLC and
+// other Serac Project Developers. See the top-level LICENSE file for
+// details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+
+/**
+ * @file solid_functional_material.hpp
+ *
+ * @brief The material and load types for the solid functional physics module
+ */
+
+#pragma once
+
+#include "serac/numerics/functional/functional.hpp"
+#include "serac/physics/materials/solid_functional_material.hpp"
+
+/// SolidFunctional helper data types
+namespace serac::solid_util {
+
+/**
+ * @brief Linear isotropic elasticity material model
+ *
+ * @tparam dim Spatial dimension of the mesh
+ */
+template <int dim>
+class ParameterizedLinearIsotropicSolid {
+public:
+  /**
+   * @brief Construct a new Linear Isotropic Elasticity object
+   *
+   * @param density Density of the material
+   * @param shear_modulus_offset Shear modulus offset of the material
+   * @param bulk_modulus_offset Bulk modulus offset of the material
+   */
+  ParameterizedLinearIsotropicSolid(double density = 1.0, double shear_modulus_offset = 1.0,
+                                    double bulk_modulus_offset = 1.0)
+      : density_(density), bulk_modulus_offset_(bulk_modulus_offset), shear_modulus_offset_(shear_modulus_offset)
+  {
+    SLIC_ERROR_ROOT_IF(density_ < 0.0, "Density must be positive in the linear isotropic elasticity material model.");
+  }
+
+  /**
+   * @brief Material response call for a linear isotropic solid
+   *
+   * @tparam T1 Spatial position type
```

Comment:
As someone potentially using this API, I'd find it easier to understand and use if the `T1` through `T5` parameters were replaced with, e.g., `SpatialPositionType`, particularly if the intent is for users to emulate this pattern.

Reply:
I'm OK with either. Do you have an opinion @samuelpmishLLNL @btalamini @white238?

Reply:
I just think this convention should be consistent. I followed the `T` style after tensor/functional, but the types here definitely have a physical meaning.

Reply:
I remember @tepperly having a similar question, so I think two potential users bring this up means we should change it.

Reply:
I think it does make sense to use more descriptive template parameter names here, since these quantities aren't so abstract. For some of the lower-level types (e.g. `tensor`, `tuple`, etc) those template parameters can represent practically anything (so it's harder to attach a descriptive name), whereas these ones are much more specific.

Also, we only need to use a template parameter for arguments that might be promoted to AD-enabled versions, so I think the spatial position could just be `tensor< double, dim > x`, rather than a template parameter `SpatialPositionType x`.

The naming of arguments here is also a little inconsistent, we're using `x`, `du_dX` (mathematical notation) for some argument, and then `displacement`, `bulk_parameter`, `shear_parameter` (descriptive words) for the others. I'm not sure which is better, maybe a compromise like:

```cpp
  template <typename displacement_type, 
            typename displacement_gradient_type, 
            typename bulk_modulus_type, 
            typename shear_modulus_type >
   SERAC_HOST_DEVICE auto operator()(const tensor< double, dim > & /* x */, 
                                     const displacement_type & /* u */, 
                                     const displacement_gradient_type & du_dX, 
                                     const bulk_modulus_type & K, 
                                     const shear_modulus_type & G) const { ... }
```

Reply:
> For some of the lower-level types (e.g. tensor, tuple, etc) those template parameters can represent practically anything (so it's harder to attach a descriptive name)

I agree. I'm not terribly concerned about terse template arguments for low-level, algebraic data types; I'm mostly concerned with arguments that have physical significance.

re: `x`, `u` versus `position`, `displacement`, as someone without a solid mechanics background, I find the latter more helpful due to collisions in notation namespaces for different disciplines (e.g., `mu` could be a shear modulus, chemical potential, coefficient of static friction, or equality-constraint duality multiplier, depending on context). The longer names are also easier to `grep` when making changes, easier to Google if I'm unfamiliar with the physics, and less likely to become stale than explanatory comments.